src/Engine/Renderer/Mesh/Mesh.inl

```diff
@@ -464,10 +467,62 @@ PointCloud::PointCloud( const std::string& name,
 }
 
 /////////  LineMesh ///////////
+
 LineMesh::LineMesh( const std::string& name,
                     typename base::CoreGeometry&& geom,
                     typename base::MeshRenderMode renderMode ) :
     base( name, std::move( geom ), renderMode ) {}
 
+/////////  PolyMesh ///////////
+
+void PolyMesh::updateGL_specific_impl() {
+    if ( !m_indices )
+    {
+        m_indices      = globjects::Buffer::create();
+        m_indicesDirty = true;
+    }
+    if ( m_indicesDirty )
+    {
+        triangulate();
+        /// this one do not work since m_indices is not a std::vector
+        // m_indices->setData( m_mesh.m_indices, GL_DYNAMIC_DRAW );
+        m_numElements = m_triangleIndices.size() * Core::Vector3ui::RowsAtCompileTime;
+
+        m_indices->setData(
+            static_cast<gl::GLsizeiptr>( m_triangleIndices.size() *
+                                         sizeof( typename base::CoreGeometry::IndexType ) ),
+            m_triangleIndices.data(),
+            GL_STATIC_DRAW );
+        m_indicesDirty = false;
+    }
+    if ( !base::m_vao ) { base::m_vao = globjects::VertexArray::create(); }
+    base::m_vao->bind();
+    base::m_vao->bindElementBuffer( m_indices.get() );
+    base::m_vao->unbind();
+}
+
+void PolyMesh::triangulate() {
+    m_triangleIndices.clear();
+    m_triangleIndices.reserve( m_mesh.getIndices().size() );
+    for ( const auto& face : m_mesh.getIndices() )
+    {
+        LOG( logDEBUG ) << face.cols() << " " << face.rows() << " " << face.size();
+
+        if ( face.size() == 3 ) { m_triangleIndices.push_back( face ); }
+        else
+        {
+            auto i0 {face[0]};
+            auto i1 {face[1]};
+            /// simple fan type triangulation
```

Comment:
Should be better to use sewing triangulation, that is the same as fan one for polygon with less than 
5 vertices but will prevent rendering artifacts for polygons with more than 6 vertices while being as simpler as fan triangulation for convex planar polygons.
![IMG_1340](https://user-images.githubusercontent.com/11328578/87122934-1a3f8c80-c286-11ea-9d82-97ffbe430797.jpg)

Reply:
Do you have a specific algorithm in mind ?

Reply:
Yes, I draw it in red on the picture above. It is really very similar to the fan algo 
Instead of generating list of visited vertices (and emit triangle at each visited vertice starting at the third) in a fan mode (e.g 0123456 in the above example), you generate it in a strip mode (0126354) and emit a triangle at for each visited vertex starting at the third. Then, the only difference between the two is just the order of the visited vertices.
Not so clear if explained like this but I can implement it rather quickly (well, it might take 1 hour if if find such an hour in my agenda :p )

Reply:
That's fine I'll do it

Reply:
first version done, but I'll need to simplify it.

Reply:
35 min left for the "one hour implementation". You're in time. Run @dlyr, run :D

Reply:
well I've finish my second one...

Reply:
But still want to get rid of one index ...

Reply:
There is some kind of related algorithm used when removing an edge in a tetrahedral mesh, which is done by dynamic programming (see paper `Minimal  Triangulations  of  Polygonal  Domains`)

Reply:
I have pushed the two (not tested as everything in the draft PR) versions